R/g.plot5.R

```diff
@@ -91,60 +100,79 @@ g.plot5 = function(metadatadir = c(), dofirstpage = FALSE, viewingwindow = 1, f0
           }
           if (length(della) > 0) summarysleep_tmp = summarysleep_tmp[-della,]
         }
-        # do not include days with no meaningful data
         threshold_hrs_of_data_per_day = 0.5
-        d2excludeb = d2exclude = which(daysummary_tmp$N.valid.hours < threshold_hrs_of_data_per_day)
-        n2excludeb = n2exclude = which(summarysleep_tmp$fraction_night_invalid > 0.66
-                                       | summarysleep_tmp$SptDuration == 0)
-        if (length(d2exclude) > 0) {
-          d2excludeb = daysummary_tmp$measurmentday[d2exclude]
-          daysummary_tmp = daysummary_tmp[-d2exclude,] #ignore days with non-wear
-          d2exclude = d2excludeb
-        }
-        
-        if (length(tail_expansion_log) != 0) { # then keep timing of sleeponset to plot
-          lastnight = max(summarysleep_tmp$night)
-          if (lastnight %in% n2exclude) {
-            n2excludeb = n2exclude = n2exclude[-which(n2exclude == lastnight)]
+        if (visualreport_without_invalid == TRUE) {
+          incrementNight = 0
+          # do not include days with no meaningful data
+          d2excludeb = d2exclude = which(P2daysummary_tmp$N.valid.hours < threshold_hrs_of_data_per_day)
+          n2excludeb = n2exclude = which(summarysleep_tmp$fraction_night_invalid > 0.66
+                                         | summarysleep_tmp$SptDuration == 0)
```

Comment:
Should this 0.66 be `includenightcrit`?

Reply:
Good point, and I was also not using `includedaycrit`. I have now revised the code to use both and I have clarified the minimum per 0.5 hours inside the code as well.